File: templates/typescript/template.ts
import { parse } from "https://deno.land/std@0.119.0/flags/mod.ts";

let debugEnabled = false;

function debug(out: string) {
  if (debugEnabled) {
    console.log(out);
  }
}

// ----------------------------- SOLUTION CODE -----------------------------
function solve(data: string): string {
  // TODO: solve
  return "";
}
// --------------------------- END SOLUTION CODE ---------------------------

async function main() {
  const flags = parse(Deno.args, {
    boolean: ["d"],
    string: ["f"],
    default: {
      debug: false,
    },
    alias: {
      d: "debug",
      f: "filename",
    },
  });

  const { debug, filename } = flags;
  debugEnabled = debug;
  const data = await Deno.readTextFile(filename);

  const solution = solve(data);
  console.log(solution);
}

await main();

File: templates/python/template.py
from argparse import ArgumentParser

debug_enabled = False


def debug(out: str):
    if debug_enabled:
        print(out)


# ----------------------------- SOLUTION CODE -----------------------------


def solve(data: str) -> str:
    # TODO: solve
    pass


# --------------------------- END SOLUTION CODE ---------------------------


def main():
    parser = ArgumentParser(description="Solve AoC puzzles")
    parser.add_argument("filename")
    parser.add_argument("-d", "--debug", action="store_true", default=False)

    args = vars(parser.parse_args())
    if not args:
        parser.print_help()
    else:
        file_name = args["filename"]
        global debug_enabled
        debug_enabled = args["debug"]

        with open(file_name) as data:
            solution = solve(data.read())
            print(solution)


if __name__ == "__main__":
    main()
